app/src/main/java/org/jellyfin/androidtv/ui/playback/rewrite/RewriteMediaManager.kt

```diff
@@ -0,0 +1,282 @@
+package org.jellyfin.androidtv.ui.playback.rewrite
+
+import android.content.Context
+import android.widget.Toast
+import androidx.lifecycle.ProcessLifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.jellyfin.androidtv.constant.QueryType
+import org.jellyfin.androidtv.preference.UserSettingPreferences
+import org.jellyfin.androidtv.ui.itemhandling.BaseRowItem
+import org.jellyfin.androidtv.ui.itemhandling.ItemRowAdapter
+import org.jellyfin.androidtv.ui.navigation.Destinations
+import org.jellyfin.androidtv.ui.navigation.NavigationRepository
+import org.jellyfin.androidtv.ui.playback.AudioEventListener
+import org.jellyfin.androidtv.ui.playback.MediaManager
+import org.jellyfin.androidtv.ui.playback.PlaybackController
+import org.jellyfin.androidtv.ui.presentation.CardPresenter
+import org.jellyfin.playback.core.PlaybackManager
+import org.jellyfin.playback.core.model.PlayState
+import org.jellyfin.playback.core.queue.Queue
+import org.jellyfin.playback.core.queue.item.QueueEntry
+import org.jellyfin.playback.core.queue.queue
+import org.jellyfin.playback.jellyfin.queue.item.BaseItemDtoUserQueueEntry
+import org.jellyfin.sdk.api.client.ApiClient
+import org.jellyfin.sdk.model.api.BaseItemDto
+import kotlin.time.Duration.Companion.milliseconds
+
+@Suppress("TooManyFunctions")
```

Comment:
Is there a way we can break this down using composition?

I'm thinking if you had multiple instances such as `AudioManager`, `VideoManager`, `CodecManager` ....etc. then this simply becomes a glue class that delegates to it's instances. It also makes it trivial to structure future code too

Even if it's a single line currently, it means when we inevitably add more and more code it's already encapsulated outside of this class

I'm concerned that whilst this is a good improvement, there's a lot of implementation in a single location still and `Media` turns into a catch-all term

Reply:
This is just an implementation for an interface that will be removed when we rewrite the UI. Just bridging the new code to the old spaghetti here.